pyomo/contrib/pynumero/interfaces/nlp.py

```diff
@@ -1881,3 +1722,59 @@ def constraint_idx(self, constraint):
             raise RuntimeError("Constraint must be not indexed")
         return self._conToIndex[constraint]
 
+
+import pyomo.environ as aml
```

Comment:
This looks like testing code and should be moved to a `tests` directory.  In particular, we should be cautious about "import pyomo.environ" in any Pyomo code outside of examples and tests (it leads to circular imports)

Reply:
Removed